core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStore.java

```diff
@@ -106,12 +107,19 @@ public KafkaStore(SchemaRegistryConfig config,
     this.zkUtils = ZkUtils.apply(
         kafkaClusterZkUrl, zkSessionTimeoutMs, zkSessionTimeoutMs,
         JaasUtils.isZkSecurityEnabled());
-    this.kafkaTopicReader =
-        new KafkaStoreReaderThread<>(zkUtils, kafkaClusterZkUrl, topic, groupId,
-                                         Integer.MIN_VALUE, this.storeUpdateHandler,
-                                         serializer, this.localStore, this.noopKey);
     this.brokerSeq = zkUtils.getAllBrokersInCluster();
 
+    List<Broker> brokers = JavaConversions.seqAsJavaList(brokerSeq);
+    StringBuilder sb = new StringBuilder();
+    for (Broker broker : brokers) {
+      for(EndPoint ep : JavaConversions.asJavaCollection(broker.endPoints().values())) {
```

Comment:
Might not want to throw an exception -- some folks might want to use both secure and plaintext listeners, even if not everything has migrated over to secure access yet. Throwing an exception would break this for them without a way to override it. It might be better to filter out secure endpoints, log a warning, and throw an exception if you end up with an empty list.